firmware/raspi/gui.py

```diff
@@ -0,0 +1,353 @@
+            # _, human_plays_white_pieces, board, ai_player = params
+
+        self.players = params["players"]
+        if params["human_plays_white_pieces"]:
+            self.turn = 0
+            self.move_first = "gui"
+        else:
+            self.turn = 1
+            self.move_first = "ai"
+          
+        if not self.turn:
+            print("Please make your move on the board.")
+        else:
+            self.game.process(self.players[0])
+            self.chessboard.show(self.game.current_fen())
+            self.draw_board()
+            self.show_move(self.game.last_made_move(), self.game.is_white_turn())
+            self.draw_pieces()
+            self.turn = 1
+        self.parent.after(10, self.update_state)
+        self.parent.mainloop()
+
+    def update_state(self):
+        """Runs simultaneously with the tkinter main loop and checks if a move has been
+        made every 10ms 
+        """
+        made_move = self.game.process(self.players[self.turn])
+        if made_move:
+            self.selected_piece = None
+            self.focused = None
+            self.pieces = {}
+            if (self.turn and self.move_first == "gui") or (not self.turn and self.move_first == "ai"):
+                self.chessboard.show(self.game.current_fen())
+                self.show_move(self.game.last_made_move(), self.game.is_white_turn())
+            self.draw_board()
+            self.draw_pieces()
+            self.turn = (self.turn + 1) % 2
+        self.parent.after(10, self.update_state)
+
+    def square_clicked(self, event):
+        """Waits for click on the board and depending on player's turn, executes the move
+        """
+        if not self.turn or (self.turn and self.move_first == "ai"):
+            col_size = row_size = self.dim_square
+            selected_column = int(event.x / col_size)
+            selected_row = 7 - int(event.y / row_size)
+            pos = self.chessboard.alpha_notation((selected_row, selected_column))
+            try:
+                _ = self.chessboard[pos]
+            except Exception as e:
+                pass
+            if self.selected_piece:
+                self.made_move = self.shift(self.selected_piece[1], pos)
+                if self.made_move == None:
+                    self.selected_piece = None
+                    self.focus(pos)
+                    self.draw_board()
+                    print("Invalid Move! Choose a new piece.")
+                    return
+                self.players[self.turn].set_move(self.made_move)
+            self.focus(pos)
+            self.draw_board()
+
+        if self.game.is_game_over():
+            print("Checking if game is over...")
+            # Need to empty move queue to play out final captures, etc (if any) before ending
+            while self.game.board.msg_queue:
+                self.game.process(None)
+            #gui player can no longer make moves
```

Comment:
Can you please update all your comments to have a space between the # and the beginning of the comment? And please capitalize the first word of the comments.